src/java/org/apache/cassandra/index/sai/disk/v2/hnsw/CassandraOnDiskHnsw.java

```diff
@@ -128,7 +129,10 @@ public VectorPostingList search(float[] queryVector, int topK, int limit, Bits a
     @Override
     public VectorPostingList search(float[] queryVector, int topK, float threshold, int limit, Bits bits, QueryContext context)
     {
-        throw new UnsupportedOperationException();
+        if (threshold > 0)
+            throw new InvalidRequestException("Geo queries are not supported for legacy SAI indexes -- drop the index and recreate it to enable these");
```

Comment:
This will fail on the replica and will not make it back to the client. I am not sure what is available to us, but we'll want to build on https://github.com/datastax/cassandra/pull/852.